src/systems/systems/src/navigation/radio_altimeter.rs

```diff
@@ -0,0 +1,722 @@
+    }
+}
+
+impl TransceiverPair for Ala52BTransceiverPair {
+    /// Returns the time the signal took to travel from one of the transceivers to the other,
+    /// to the receiver, or None if there is no path (or the signal is too weak).
+    /// While it is simplified compared to a proper frequency simulation, you can imagine that the
+    /// measured runtime is already based on the filtered difference between the sent and received
+    /// frequency in an FMCW-based radar.
+    fn response(&self) -> Option<TransceiverPairResponse> {
+        if self.pitch.abs() > Angle::new::<degree>(45.)
+            || self.bank.abs() > Angle::new::<degree>(44.)
+        {
+            return None;
+        }
+
+        // First, we need to determine the shortest path of the radio waves between the two
+        // transceivers. This will usually be based on a reflection on the ground, but one future
+        // failure case might be direct coupling where e.g. contamination leads to the shortest path
+        // being radio waves traveling directly along the fuselage, leading to extremely low
+        // readings.
+
+        // As preparation, calculate the perpendicular distance from both transceivers to the
+        // ground.
+        let a: Length = self.transmitter_height_over_ground();
+        let b: Length = self.receiver_height_over_ground();
+
+        // Perform some 2D geometry to determine the shortest path between the two transceivers and
+        // the ground. The basic idea is that, given two transceivers A, B and a line g representing
+        // the ground. We now would like to determine the length of the shortest path from A to B
+        // via a point on the line. To do this, you can can reflect one of the transceivers B
+        // along the ground g to construct B'. The intersection of the line A-B' with g will result
+        // in the reflection point C in g, and this line A-C-B' now is equal to the shortest path
+        // A-C-B. By then calculating some right angles between A, B', and g, we eventually have a
+        // simple pythagorean triangle where the hypotenuse corresponds to A-C-B', which we can
+        // calculate trivially.
+        let transceiver_along_ground_distance = self.antenna_along_ground_distance();
+        let path_in_air: Length = ((a + b) * (a + b)
+            + transceiver_along_ground_distance * transceiver_along_ground_distance)
+            .sqrt();
+        let shortest_path_length: Length =
+            path_in_air + self.transmitter.cable_length + self.receiver.cable_length;
+
+        // At this point we've determined the length of the shortest path between the two
+        // transceivers. We now convert it into a travel time.
+
+        // It would be nice if this could be a const, but unfortunately the Velocity constructor is
+        // too complicated for what Rust deems acceptable for a const.
+        let speed_of_light: Velocity = Velocity::new::<meter_per_second>(299_792_458.);
```

Comment:
If we want to nitpick the propagation velocity in the cables will be approx 2/3 * c for typical coax. The effect will be too small to be significant though when airborne.😁